data/actionsModels.js

```diff
@@ -0,0 +1,36 @@
+const knex = require('knex');
+const dbConfig = require('../knexfile');
+const db = knex(dbConfig.development);
+
+function get(id){
+    if(id){
+        return db('actions').where('id', id).then(action => { return action[0] })
+    }
+
+    return db('actions').then(actions => { return actions })
+}
+
+function add(action){
+    return db('actions').insert(action).then( ([id]) => this.get(id) )
```

Comment:
Your callback here is clever; I like it (but it took me a second to see that you were using the `this` of your exports object)! I'm not sure if you would want to do this in production code, though. Since the get is already defined in the global scope, you could simply do:
```javascript
return db('actions').insert(action).then(([ id ]) => get(id))
```